### PR TITLE
Doc updates

### PR DESCRIPTION
**Pull request by codenrhoden**
A few general doc cleanup items. Mostly things I noticed while browsing https://image-builder.sigs.k8s.io

/assign @detiber

### CODE REVIEW

File: hack/build-mdbook.sh [hunk 1]
```diff
@@ -1,64 +0,0 @@
-#!/bin/bash
```
Comment (detiber) [inline hack/build-mdbook.sh]: It looks like this file is still called by the Makefile, should it be calling `docs/book/build.sh` instead similar to the netlify config?

Comment (codenrhoden) [reply hack/build-mdbook.sh]: ah, good catch on the Makefile. Yeah, I'll fix that!

**Comment by codenrhoden** [comment]
@detiber Makefile is updated!

**Pull request merged**